src/Functions/formatRow.cpp

```diff
@@ -0,0 +1,102 @@
+#include <memory>
+#include <Columns/ColumnString.h>
+#include <DataStreams/materializeBlock.h>
+#include <DataTypes/DataTypeString.h>
+#include <Formats/FormatFactory.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+#include <IO/WriteBufferFromVector.h>
+#include <IO/WriteHelpers.h>
+#include <Processors/Formats/IOutputFormat.h>
+#include <ext/map.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+
+/** formatRow(<format>, x, y, ...) is a function that allows you to use RowOutputFormat over
+  * several columns to generate a string per row, such as CSV, TSV, JSONEachRow, etc.
+  */
+
+class FunctionFormatRow : public IFunction
+{
+public:
+    static constexpr auto name = "formatRow";
+
+    FunctionFormatRow(const String & format_name_, const Context & context_) : format_name(format_name_), context(context_) { }
+
+    String getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 0; }
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0}; }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) override
+    {
+        auto col_str = ColumnString::create();
+        ColumnString::Chars & vec = col_str->getChars();
+        WriteBufferFromVector buffer(vec);
+        ColumnString::Offsets & offsets = col_str->getOffsets();
+        offsets.resize(input_rows_count);
+        Block arg_block;
+        for (auto i = 1u; i < arguments.size(); ++i)
+            arg_block.insert(block.getByPosition(arguments[i]));
+        materializeBlockInplace(arg_block);
+        auto out = FormatFactory::instance().getOutputFormat(format_name, buffer, arg_block, context, [&](const Columns &, size_t row)
+        {
+            *buffer.position() = '\0'; // replace '\n' with '\0'
```

Comment:
It doesn't look consistent enough.
Maybe just keep all characters including trailing newline... or make two functions so the second one will be also appropriate for RowBinary, Native, etc.

Reply:
`formatRowWithoutNewline` and `formatRow`?

Reply:
Added a new function `formatRowNoNewline`